src/tests/Microsoft.Agents.BotBuilder.Dialogs.Tests/DialogContextTests.cs

```diff
@@ -0,0 +1,167 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Globalization;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Agents.Core.Interfaces;
+using Microsoft.Agents.Core.Models;
+using Moq;
+using Xunit;
+
+namespace Microsoft.Agents.BotBuilder.Dialogs.Tests
+{
+    public class DialogContextTests
+    {
+        private readonly Mock<ITurnContext> _context = new();
+        private readonly DialogContext _dialogContext;
+
+        public DialogContextTests()
+        {
+            _dialogContext = new(new DialogSet(), _context.Object, new DialogState());
+        }
+
+        [Fact]
+        public async Task BeginDialogAsync_ShouldThrowOnNullDialogId()
+        {
+            await Assert.ThrowsAsync<ArgumentNullException>(() => _dialogContext.BeginDialogAsync(null));
+        }
+
+        [Fact]
+        public async Task BeginDialogAsync_ShouldThrowOnEmptyDialogId()
+        {
+            await Assert.ThrowsAsync<ArgumentNullException>(() => _dialogContext.BeginDialogAsync(string.Empty));
+        }
+
+        [Fact]
+        public async Task BeginDialogAsync_ShouldThrowOnNullDialog()
```

Comment:
I think this one could be renamed.
```suggestion
        public async Task BeginDialogAsync_ShouldThrowOnInexistingDialog()
```